File: 프로그래머스/lv2/카펫.py
def solution(brown, yellow):
    answer = []
    for i in range(1,int(yellow**0.5)+1):
        if yellow%i==0:
            #가로가 세로보다 더김, 노란색의 가로, 세로 길이 구함
            a,b=yellow//i,i
            #갈색은 노란색 테두리에 있으니까 양옆 붙고 네모 4칸이 남으니 4 더해줌
            if (a+b)*2+4==brown:
                answer=[a+2,b+2]
    return answer

print(solution(24,24))

File: 프로그래머스/lv2/카펫.java
public class 카펫 {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        for (int i=1; i<=Math.sqrt(yellow)+1; i++) {
            if (yellow%i==0){
                int a=yellow/i;
                int b=i;
                if ((a+b)*2+4==brown){
                    answer[0]=a+2;
                    answer[1]=b+2;
                    break;
                }
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        카펫 carpet = new 카펫();
        int[] answer=carpet.solution(8,1);
        System.out.println(answer[0]+" "+answer[1]);
    }
}
